File: php/src/problem0001/twoSum_test.php
<?php

namespace problem0001;

$solution = new Solution();

dataset("test", [
    [[0, 1], [2, 7, 11, 15], 9],
    [[1, 2], [3, 2, 4], 6],
    [[0, 1], [3, 3,], 6],
]);

test("1.两数之和", function (array $expected, array $nums, int $target) use ($solution): void {
    expect($expected)->toEqual($solution->twoSum($nums, $target));
})->with("test");

File: golang/src/problem0001/twoSum_test.go
package problem0001

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type Case struct {
	expected []int
	nums     []int
	target   int
}

func TestTwoSum(t *testing.T) {
	cases := [3]Case{
		{[]int{0, 1}, []int{2, 7, 11, 15}, 9},
		{[]int{1, 2}, []int{3, 2, 4}, 6},
		{[]int{0, 1}, []int{3, 3}, 6},
	}

	for _, c := range cases {
		t.Run("1.两数之和", func(t *testing.T) {
			assert.Equal(t, c.expected, twoSum(c.nums, c.target))
		})
	}
}
